model/src/appdata_hexadecimal.rs

```diff
@@ -1,26 +1,27 @@
+use crate::order::AppId;
 use serde::{de, Deserializer, Serializer};
 use std::fmt;
 
-pub fn serialize<S>(value: &[u8; 32], serializer: S) -> Result<S::Ok, S::Error>
+pub fn serialize<S>(value: &AppId, serializer: S) -> Result<S::Ok, S::Error>
```

Comment:
Now that AppId is its own type that we control we `impl Serialize` and Deserialize on it directly without needing this module. The serialization can use FromStr and Display so we don't have to juggle with bytes anymore here.
The serde traits are a bit hairy so I'm fine merging as is too and tackling that later.

Reply:
I am not sure I entirely follow what would change about this. Like delete this file completely and implement the same stuff somewhere else?

Reply:
> The serde traits are a bit hairy so I'm fine merging as is too and tackling that later.

But the `serialize` and `deserialize` methods are basically the implementations for `Serialize::serialize` and `Deserialize::deserialize` right? So it would just mean wrapping these existing implementations in:
```rust
impl Serialize for AppId {
  fn serialize(...) -> ... {
    // the current implementation in this function...
  }
}
```

And do the same for Deserialize.

>  Like delete this file completely and implement the same stuff somewhere else?

It might be worth creating a new module `model/src/appid.rs` with the definition of `AppId` as well as the `FromStr`, `{Des,S}erialize`, etc.

Reply:
Done!